src/composables/websocket.ts

```diff
@@ -0,0 +1,192 @@
+import { v4 as genUuid } from 'uuid';
+import { ref, watch } from '@vue/composition-api';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import {
+  WEB_SOCKET_CHANNEL_KEY_BLOCKS,
+  WEB_SOCKET_CHANNEL_MICRO_BLOCKS,
+  WEB_SOCKET_CHANNEL_TRANSACTIONS,
+  handleUnknownError,
+} from '../popup/utils';
+import type {
+  IDefaultComposableOptions,
+  IMiddlewareWebSocketSubscriptionMessage,
+  INetwork,
+  WebSocketChannelName,
+} from '../types';
+import { useGetter } from './vuex';
+
+let wsClient: WebSocket;
+const isWsConnected = ref(false);
+// eslint-disable-next-line no-unused-vars
+const subscribers: Record<WebSocketChannelName, Record<string, (payload: any) => void>> = {
+  Transactions: {},
+  MicroBlocks: {},
+  KeyBlocks: {},
+};
+
+const channelsSubscribed: Record<WebSocketChannelName, boolean> = {
+  Transactions: false,
+  MicroBlocks: false,
+  KeyBlocks: false,
+};
+
+// eslint-disable-next-line no-unused-vars
+export function useWebSocket({ store }: IDefaultComposableOptions) {
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+
+  /**
+   * This function is used to subscribe to a specific WebSocket channel.
+   * @param {WebSocketChannelName} channelName - The name of the channel to subscribe to
+   */
+  function subscribeToWebSocketChannel(channelName: WebSocketChannelName) {
+    if (
+      !channelName
+      || channelsSubscribed[channelName]
+      || !isWsConnected.value
+    ) {
+      return;
+    }
+
+    const message: IMiddlewareWebSocketSubscriptionMessage = {
+      op: 'Subscribe',
+      payload: channelName,
+    };
+
+    wsClient.send(JSON.stringify(message));
+    channelsSubscribed[channelName] = true;
+  }
+
+  /**
+   * Handle websocket message event.
+   * @param {MessageEvent} message - The message event object.
+   */
+  function handleWebsocketMessage(message: MessageEvent) {
+    if (!message.data) {
+      return;
+    }
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers[data.subscription as WebSocketChannelName]
+      ) {
+        return;
+      }
+      // Call all subscribers for the channel
+      Object.values(subscribers[data.subscription as WebSocketChannelName]).forEach(
+        (subscriberCb) => subscriberCb(data.payload),
+      );
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  function subscribeToWebSocketChannels() {
+    wsClient.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    (Object.keys(channelsSubscribed) as WebSocketChannelName[]).forEach(
+      (channelName) => subscribeToWebSocketChannel(channelName),
+    );
+  }
+
+  /**
+   * UnSubscribes websocket channels.
+   */
+  function unsubscribeFromWebSocketChannels() {
+    // Remove event listener for messages
```

Comment:
I think it's obvious what the below line does - I suggest to remove this comment.